File: mpv-unlimit-random/unlimit-random.lua
local random = false

function random_play()
	if random then
		plct = mp.get_property("playlist-count")
		pos = mp.get_property("playlist-pos-1")
		mp.command("playlist-shuffle")
		while (pos == plct) do
			mp.command("playlist-shuffle")
			pos = mp.get_property("playlist-pos-1")
		end
	end
end
function random_play_control()
	random = not random
	mp.osd_message("Random: " .. (random and "yes" or "no"))
end

mp.register_event("start-file", random_play)
mp.add_key_binding('y', "random_control", random_play_control)

File: mpv-unlimit-random/unlimit-random.js
'use_strict'

var random=false;

function random_play() {
	var pl,pos;
	if (random) {
		pl=mp.get_property_number('playlist-count');
		do {
			mp.command('playlist-shuffle');
			pos=mp.get_property_number('playlist-pos-1');
		} while(pl===pos);
	}
}
function random_play_control() {
	random=!random;
	mp.osd_message('Random: '+(random?'yes':'no'));
}

mp.register_event('start-file',random_play)
mp.add_key_binding('y','random_control',random_play_control)